tests/Unit/Entity/BannerSelection/CampaignTest.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace WMDE\BannerServer\Tests\Unit\Entity\BannerSelection;
+
+use WMDE\BannerServer\Entity\BannerSelection\Campaign;
+use WMDE\BannerServer\Entity\BannerSelection\TrafficImpressionThreshold;
+
+class CampaignTest extends \PHPUnit\Framework\TestCase {
+
+	public static function returnRandomBucket() {
+		return '';
+	}
+
+	public function testGivenTimeOutOfDateRange_campaignIsNotActive() {
+		$campaign = new Campaign(
+			'C18_WMDE_Test',
+			new \DateTime( '2018-10-01' ),
+			new \DateTime( '2018-10-31' ),
```

Comment:
This test is perfectly valid, but reminds me that when you're writing the parser you should either require exact dates with times (and make sure that the timezone is set) or choose different default times for start and end date. A day always starts at 0:00:00 and always ends at 23:59:59.